app.rb

```diff
@@ -0,0 +1,138 @@
+require './person'
+require './student'
+require './rental'
+require './classroom'
+require './teacher'
+
+class App
+  def initialize
+    @persons = []
+    @books = []
+    @rentals = []
+  end
+
+  def home
+    puts 'Welcome to School Library Application'
+    until options_list
+      input = gets.chomp.to_i
+      if input == 7
+        puts 'Thanks for using School Library...'
+        exit
+      end
+
+      option input
+    end
+  end
+
+  def list_persons
+    puts 'Person list is empty. Please add a person to your list.' if @persons.empty?
+    @persons.each do |person|
+      puts "Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def list_books
+    puts 'List of Books is empty. Please add a Book to your ist.' if @books.empty?
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def add_person
+    puts 'Who do you want to add?'
+    puts '1. Student'
+    puts '2. Teacher'
+    input = gets.chomp.to_i
+    if input == 1
+      add_student
+    elsif input == 2
+      add_teacher
+    else
+      puts 'Invalid Option Input. Retry'
+    end
+  end
+
+  def add_student
+    puts 'Adding Student..'
+    puts 'Please enter Name: '
+    name = gets.chomp
+    puts 'Please enter Age: '
+    age = gets.chomp.to_i
+    puts 'Had Parent Permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'y'
+      student = Student.new(age, name, parent_permission: true)
+      @persons << student
+      puts 'Student Added Successfully with Partent Permission'
+    when 'n'
+      student = Student.new(age, name, parent_permission: false)
+      @persons << student
+      puts 'Student Added Successfully but Without Parent Permission'
+    else
+      puts 'Invalid Option Input. Retry'
+      add_student
+    end
+  end
+
+  def add_teacher
+    puts 'Adding Teacher'
+    puts 'Please enter Age: '
+    age = gets.chomp.to_i
+    puts 'Please enter Name: '
+    name = gets.chomp
+    puts 'Please enter Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(name, age, specialization)
+    @persons << teacher
+    puts "Teacher Added successfully witn ID #{teacher.id}"
+  end
+
+  def add_book
+    puts 'Adding Book'
+    puts 'Please enter Book Title: '
+    title = gets.chomp
+    puts 'Please enter Book Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @book.push(book)
+    puts "Book #{title} added successfully"
+  end
```

Comment:
- [ ] An error is thrown the like below when attempting to create a book. Please ensure to improve upon the same.
![image](https://github.com/Cosmosha/school_library/assets/12745474/7c75a022-a12b-42d4-a403-5c0fd9e116f6)

Reply:
Thank you @rloterh for the feedback, the errors has been fixed.